src/Cryptol/TypeCheck/SimpType.hs

```diff
@@ -113,6 +113,7 @@ tSub :: Type -> Type -> Type
 tSub x y
   | Just t <- tOp TCSub (op2 nSub) [x,y] = t
   | tIsInf y  = tError (tf2 TCSub x y)
```

Comment:
If we assume the precondition that the second argument (`y`) is finite, then this case must be unreachable, right?